# Bài làm thêm 1 - Numpy



In [1]:
import numpy as np

In [2]:
#1. Đọc dữ liệu từ tập tin shark.tsv (dữ liệu số vụ cá mập tấn công các tháng của 3 quốc gia USA,Australia,South Africa)
# và đưa vào mảng shark
shark = np.genfromtxt('shark.tsv',delimiter='\t',skip_header=1,dtype=int)
print('shark', shark)

shark [[ -1  28  94  68]
 [ -1  27  78  32]
 [ -1  66  63  34]
 [ -1 103  54  25]
 [ -1 106  21  25]
 [ -1 173  25  22]
 [ -1 232  25  28]
 [ -1 208  28  15]
 [ -1 182  35  17]
 [ -1 141  56  16]
 [ -1  79  65  20]
 [ -1  37  94  46]]


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[ -1,  28,  94,  68],
       [ -1,  27,  78,  32],
       [ -1,  66,  63,  34],
       [ -1, 103,  54,  25],
       [ -1, 106,  21,  25],
       [ -1, 173,  25,  22],
       [ -1, 232,  25,  28],
       [ -1, 208,  28,  15],
       [ -1, 182,  35,  17],
       [ -1, 141,  56,  16],
       [ -1,  79,  65,  20],
       [ -1,  37,  94,  46]])
</pre>

</details>

In [3]:
#2. Cho biết số phần tử mỗi chiều của mảng
# data_types = np.dtype([('month', int), ('usa', int), ('aus', int), ('saf', int)])
# shark = np.array(shark, dtype=data_types)
shape = np.shape(shark)
print('shape', shape)

shape (12, 4)


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>(12, 4)
</pre>

</details>

In [4]:
#3. Thay cột 1 (giá trị -1) bằng giá trị tương ứng các tháng trong năm 1-12
shark[:,0] = np.arange(1,13)
print('shark')
print(shark)

shark
[[  1  28  94  68]
 [  2  27  78  32]
 [  3  66  63  34]
 [  4 103  54  25]
 [  5 106  21  25]
 [  6 173  25  22]
 [  7 232  25  28]
 [  8 208  28  15]
 [  9 182  35  17]
 [ 10 141  56  16]
 [ 11  79  65  20]
 [ 12  37  94  46]]


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[  1,  28,  94,  68],
       [  2,  27,  78,  32],
       [  3,  66,  63,  34],
       [  4, 103,  54,  25],
       [  5, 106,  21,  25],
       [  6, 173,  25,  22],
       [  7, 232,  25,  28],
       [  8, 208,  28,  15],
       [  9, 182,  35,  17],
       [ 10, 141,  56,  16],
       [ 11,  79,  65,  20],
       [ 12,  37,  94,  46]])
</pre>

</details>

In [15]:
#4. Hiển thị số vụ cá mập tấn công trung bình của 3 quốc gia
usa_shark_mean = shark[:, 1]
print('usa_shark_mean', usa_shark_mean)

usa_shark_mean [ 28  27  66 103 106 173 232 208 182 141  79  37]


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([115.16666667,  53.16666667,  29.        ])
</pre>

</details>

In [6]:
#5. Hiển thị số vụ cá mập tấn công trung bình của mỗi tháng  


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([63.33333333, 45.66666667, 54.33333333, 60.66666667, 50.66666667,
       73.33333333, 95.        , 83.66666667, 78.        , 71.        ,
       54.66666667, 59.        ])
</pre>

</details>

In [7]:
#6. Tạo thêm cột tổng số vụ cá mập tấn công theo mỗi tháng 
#(nếu không tính được thì đọc từ file shark1.tsv)


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[  1,  28,  94,  68, 190],
       [  2,  27,  78,  32, 137],
       [  3,  66,  63,  34, 163],
       [  4, 103,  54,  25, 182],
       [  5, 106,  21,  25, 152],
       [  6, 173,  25,  22, 220],
       [  7, 232,  25,  28, 285],
       [  8, 208,  28,  15, 251],
       [  9, 182,  35,  17, 234],
       [ 10, 141,  56,  16, 213],
       [ 11,  79,  65,  20, 164],
       [ 12,  37,  94,  46, 177]])
</pre>

</details>

In [8]:
#7. Cho biết những tháng có tổng số vụ cá mập tấn công >200


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>array([[  6, 173,  25,  22, 220],
       [  7, 232,  25,  28, 285],
       [  8, 208,  28,  15, 251],
       [  9, 182,  35,  17, 234],
       [ 10, 141,  56,  16, 213]])
</pre>

</details>

In [9]:
#8. Cho biết tổng số vụ cá mập tấn công nhiều nhất trong tháng là bao nhiêu


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>285
</pre>

</details>

In [10]:
#9. Cho biết tháng nào có tổng số vụ cá mập tấn công nhiều nhất


<details>
  <summary>Nhấn vào đây để xem kết quả !</summary>
  
  <pre>Tháng có tổng số vụ cá mập tấn công nhiều nhất là tháng : 7
</pre>

</details>